modules/03_in_memory_storage/README.md

```diff
@@ -1,9 +1,471 @@
+        """ Store a secret word."""
+        self.storage.append(secret_word)
+
+    def get_all_words(self):
+        """ Get all words saved so far. """
+        return self.storage
+```
+
+We will be able to use it like this:
+```python
+from src.in_memory_storage import InMemoryStorage
+
+storage1 = InMemoryStorage()  # create a new object of this class
+storage1.add_word("dog")
+print(storage1.get_all_words())  # prints ['dog']
+storage1.add_word("cat")
+print(storage1.get_all_words())  # prints ['dog', 'cat']
+
+storage2 = InMemoryStorage()  # create another object of this class
+print(storage2.get_all_words())  # prints empty list []
+storage2.add_word("boo!")
+print(storage2.get_all_words())  # prints ['boo!']
+```
+
+#### 2. Save words to the in-memory storage
+
+On the top of the file `api.py`, create our "database" - an object of the class InMemoryStorage.
+
+```python
+from src.in_memory_storage import InMemoryStorage
+
+database = InMemoryStorage()
+```
+
+This variable will be visible to all endpoints in `api.py`, and they all will be able to read from it
+or write to it.
+
+Change the endpoint `/upload_word` so that it uses the database:
+
+```python
+from src.in_memory_storage import InMemoryStorage
+
+database = InMemoryStorage()
+
+@app.route('/upload_word', methods=['POST'])
+def upload_word():
+    secret_word = request.form['secretWord']
+    database.add_word(secret_word)  # save word to the db
+    flash("Uploaded word " + repr(secret_word))
+    flash("All available words: " + repr(database.get_all_words()))  # display all words saved so far
+    return redirect('/')
+```
+
+Run the app, try uploading words again. Verify that now you get more informative messages,
+and that the server now remembers all the words you've uploaded so far.
+
+> Caveat: Because we store words in memory, this storage is alive as long as the server itself
+> is alive. If you shut down the server, its memory is released, and all saved words will be forgotten.
+
+> Caveat: storing words in memory also doesn't play well with running Flask in DEBUG mode.
+> Flask in DEBUG mode restarts the server every time you've changed something in the code,
+> which again means losing all the saved words.
+
+### Part 3. The Game
+
+#### Intro
+
+The game of guessing words will go as follows:
+
+- the user visits a `/game` page;
+- our server chooses a secret word the user has to guess;
+- the user sends their guesses through a form until they get it right;
+- if they get it right, they are congratulated and redirected back to the main page.
+
+Note that until the user wins the game, we need to remember the secret word that they have to guess.
+
+We did something similar when we were storing all uploaded words in the database. But this case is different:
+if multiple players visit our server, we want them to play with the same, common set of words, but their
+games shouldn't interfere. Each player should play their own game, they can't all share the secret word
```

Comment:
`Each player should play their own game, they can't all share the secret word
to guess like the share the pool of possible words.` <- I don't follow

Reply:
Looks good